### PR TITLE
Hired mercenaries

### PR DESCRIPTION
**Pull request by jafdy**
Adds fleets and "jobs" allowing merc fleets to be hired for 2 weeks. 
Price is set to 10% of aprox fleet cost. 
Most of human space is covered. (I think)

### CODE REVIEW

**Comment by LocalGod79** [comment]
i feel like these are perhaps too big - players that need this are small-time, and a 400k escort for a couple of weeks, should probably be at the *top* end of the availability...

**Comment by jafdy** [comment]
Much smaller than this and you end up with ether no real variation in size or a single interceptor. I expect that most people that use this will want to completely overwhelm any pirates so will like large fleets. 
Admittedly a fleet costing 2M to hire is a bit much. The price could be changed to 5% of value and/or the the chance of appearing could be balanced in favor of the small fleets.

**Comment by LocalGod79** [comment]
as general balance goes, you should always get more "bang for your buck" for the larger hires, so 5% might be appropriate for the top tier merc escorts (but any lower than that, and you're ranging back into purchase-on-mortgage territory).

i think that having large hireable escorts will exacerbate early game plunder/capture economy issues - since i could camp a system with a bunked-out freighter, and let my escorts disable anything that came in, return with my capture, refill my bunks, and repeat two or three times, *per hire*.

the ravening hordes aren't going to like it, but the answer is: vindictive

**Comment by jafdy** [comment]
A good point if ever I heard one.

**Comment by LocalGod79** [comment]
also, payment should be under "on accept", rather than "on offer".

the mission will not be offered if the player doesn't have enough credits to accept it.

**Comment by jafdy** [comment]
Ops, glad you spotted that, it could have been bad otherwise.

**Comment by jafdy** [comment]
Interestingly it doesn't matter weather the `payment` is in an `on offer` or `on accept` the money still gets deducted when the mission is accepted. 
The problem is that the mission can still be accepted even if you don't have enough money to pay for it.

**Comment by Amazinite** [comment]
Maybe cause it to give you a "mortgage" that you have to pay for, if that is even possible? This mortgage would then have to be payed over the course of how long you have the fleet.

**Comment by LocalGod79** [comment]
it has been my experience that the mission would not even be offered if the player didn't have enough credits for the negative payment (the same way "FW Pug 2C: Jump Drive" will not be offered if the player does not have a Jump Drive)

**Comment by jafdy** [comment]
Trouble is that this effect can trigger almost anywhere there is a shipyard. 
The cost is not carried.
And I don't know if `payment -1M` will parse, the errors file isn't throwing anything up.

**Comment by LocalGod79** [comment]
another problem to the list, so to speak.

i think that this problem is at least partially exacerbated by the value being so high.

i'm having trouble with Freighter/Bulk Freighter captains hiring the player for 60,000-150,000 for two or three jumps, while the player hires large escort fleets for no less than 400k, that hang out with them for two weeks...

i say "who cares about variety?  i want to be able to afford these escorts when i'm too poor to buy another ship!"

**Comment by jafdy** [comment]
The way iv'e been thinking about this is that you get a large amount of power almost anywhere that is expendable and much cheaper than buying a ship. If I where to stick to 10% then for 150,000 you could get 2 wasps or 3 sparrows or 1.5 hawks. If what you want is to be safe from pirates then you probably want more than a couple of fighters.

**Comment by LocalGod79** [comment]
if they're expendable, who cares if it's just a few small ships?  i just don't want *my own* to get destroyed, right?

i think that if you play test the balance of even two or three small expendable escort ships with the `heroic` personality - you might find it will achieve the desired effect of keeping the player's fleet safe...

i guess what i'm suggesting is perhaps cut the cost *and* deadline down, as well as extrapolate your jobs/fleets smaller, down to a floor that is more attainable by a trading captain who's "net worth" is under 1 or 2M credits (and actually *needs* some combat supplement, but can't really afford it)...

**Comment by jafdy** [comment]
So half the cost, half the time and a smaller even more common fleet.

**Comment by MarcelineVQ** [comment]
https://github.com/endless-sky/endless-sky/pull/1208#issuecomment-224876389
https://github.com/endless-sky/endless-sky/pull/1208#issuecomment-224937676

-1M doesn't parse but -1e6 will.
-1M can be made to parse by combining the logic of Format::Parse and DataNode::Value if it's desirable.

**Comment by jafdy** [comment]
I'm not a code person. I work with what iv'e got and try to ask for as little as possible.

File: data/jobs.txt [hunk 1]
```diff
@@ -2080,3 +2080,212 @@ mission "Marauder Hunted"
...
+		government Syndicate Republic "Free Worlds" Independent Neutral
+	to offer
+		random < 85
+	on accept
+		payment -75000
+	npc kill
+		personality vindictive heroic
+		fleet "Very Small Mercenary"
+		dialog
+			`Unfortunately the independent pilot has been killed. You try not to feel to bad about it.`
```
Comment (LocalGod79) [inline data/jobs.txt]: "...feel **too badly** about..."

**Comment by LocalGod79** [comment]
have you played a mission or trader start using these?

**Comment by jafdy** [comment]
Yes but I'm so careful that I end up having no use for them.

File: data/jobs.txt [hunk 2]
```diff
@@ -2098,7 +2098,7 @@ mission "hired mercenaries (very small)"
 		personality vindictive heroic
 		fleet "Very Small Mercenary"
 		dialog
-			`Unfortunately the independent pilot has been killed. You try not to feel to bad about it.`
```
Comment (Alkallid) [inline data/jobs.txt]: Uh, wrong one. It should be "...to feel too bad about it."

Comment (jafdy) [inline data/jobs.txt]: ops

**Comment by jafdy** [comment]
Hum. @endless-sky you do know that with out #1219 or something like it this pull contains a somewhat dramatic bug.

**Pull request closed without merging**